src/python/turicreate/toolkits/sound_classifier/sound_classifier.py

```diff
@@ -237,40 +248,29 @@ def create(dataset, target, feature, max_iterations=10,
     train_metric = mx.metric.Accuracy()
     if validation_data:
         validation_metric = mx.metric.Accuracy()
-    softmax_cross_entropy_loss = mx.gluon.loss.SoftmaxCrossEntropyLoss()
+
     for i in range(max_iterations):
         # TODO: early stopping
 
         for batch in train_data:
             data = mx.gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0, even_split=False)
             label = mx.gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0, even_split=False)
```

Comment:
This `split_and_load` stuff probably also belongs in the encapsulated class. Probably the `train` method can just take `batch.data` and `batch.label`, or perhaps `batch.data[0]` and `batch.label[0]`